nioChecksumTest/src/main/scala/cromwell/niochecksumtest/NioChecksumTest.scala

```diff
@@ -0,0 +1,123 @@
+package cromwell.niochecksumtest
+
+import akka.actor.{ActorRef, ActorSystem}
+import akka.stream.ActorMaterializer
+import cloud.nio.impl.drs.{DrsCloudNioFileSystemProvider, GoogleDrsCredentials}
+import com.typesafe.config.ConfigFactory
+import cromwell.cloudsupport.gcp.auth.UserMode
+import cromwell.core.WorkflowOptions
+import cromwell.core.io.{AsyncIo, IoCommandBuilder, IoPromiseProxyActor}
+import cromwell.engine.io.IoActor
+import cromwell.engine.io.IoActor.IoConfig
+import cromwell.filesystems.drs.{DrsPath, DrsReader}
+import cromwell.services.ServiceRegistryActor
+import net.ceedubs.ficus.Ficus.toFicusConfig
+import spray.json.JsObject
+
+import scala.concurrent.Await
+import scala.concurrent.duration.DurationInt
+import scala.util.{Failure, Success, Try}
+
+/**
+  * This class is set up to read a single (small!) DRS file into memory from our dev environment.
+  * Ensure that the Google account you're using has permission to access the DRS file in the
+  * environment you're targeting (use your test Google account in dev Terra!).
+  * (1) Log into your Google account locally: `gcloud auth login [your username]`
+  * (2) Log into Terra in your desired environment and make sure the repo you're using is linked
+  * (3) Ensure that the Martha instance below is the right one.
+  * (4) Set `drsPathToResolve` below to your DRS path
+  * (5) Update `mySecretFile` below to point to the local adc.json file for your Google account
+  * (6) "Run" NioChecksumTest using IntelliJ
+  * (7) Stop NioChecksumTest using IntelliJ because it does not stop on its own :(
+  */
+object NioChecksumTest {
```

Comment:
I realized after I was mostly done that a lot of this is a re-implementation of `DrsPathBuilderFactory`, and I may have been able to save some time by using that directly.  I don't think the guts of DPBF are relevant to what we want to test here, but if I'm wrong about that I'm happy to refactor so we use that instead.

Reply:
IMHO this seems fine for now, if we learn that DPBF code is substantially in play for this epic we can revisit that decision.